### PR TITLE
Add inital impl

### PR DESCRIPTION
**Pull request by LucioFranco**

### CODE REVIEW

File: src/cluster.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#[derive(Clone)]
+pub struct Handle {
+    event_rx: watch::Receiver<Event>,
+}
+
+pub struct Config<S, C, T, E> {
+    server: S,
+    client: C,
+    listen_target: T,
+    executor: E,
```
Comment (bIgBV) [inline src/cluster.rs]: Why be generic over the executor?

File: src/cluster.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            server,
+            client,
+            executor,
+            listen_target,
+        } = config;
+
+        let membership = Membership::new(client);
+        let mut server = server.start(listen_target).await.unwrap().fuse();
+
+        futures::select! {
```
Comment (bIgBV) [inline src/cluster.rs]: select!

**Review by bIgBV — COMMENTED** [review]
Looks like a solid start.

File: src/cluster.rs (hunk 1 above)
Comment (LucioFranco) [reply src/cluster.rs]: because i forgot to remove it!

**Pull request merged**